src/rust_connection/mod.rs

```diff
@@ -32,8 +32,8 @@ pub type GenericEvent = crate::x11_utils::GenericEvent<Buffer>;
 pub type EventAndSeqNumber = crate::connection::EventAndSeqNumber<Buffer>;
 pub type RawEventAndSeqNumber = crate::connection::RawEventAndSeqNumber<Buffer>;
 pub type BufWithFds = crate::connection::BufWithFds<Buffer>;
-pub type Error = crate::protocol::Error<Buffer>;
-pub type Event = crate::protocol::Event<Buffer>;
+pub type Error = crate::protocol::Error<GenericError>;
+pub type Event = crate::protocol::Event<GenericEvent>;
```

Comment:
Why this change?

Reply:
Oh, that's an artifact of the piece-wise way I did this. At some intermediate step this was necessary to keep things compiling.

These two type aliases are now unused. Should I just remove them or bring back their former version?

Reply:
I believe they can just be removed.

Reply:
Done